baseShip.go

```diff
@@ -41,6 +41,11 @@ func (b BaseShip) GetFuelConsumption(techs Researches, fleetDeutSaveFactor float
 	return fuelConsumption
 }
 
+// GetFuelCapacity returns ship fuel capacity
+func (b BaseShip) GetFuelCapacity() int64 {
```

Comment:
Unless ogame changed the mechanics, the amount of fuel that you can carry should increase according to your techs and character class.
Basically, the function is already implemented (`Cargo` returns you the total cargo available) -> https://github.com/alaingilbert/ogame/blob/b2ba1299e6ce5001c80ff11d643c03bf28cf6e8d/ships.go#L119-L124

Reply:
I checked the HTML JavaScript of their Calculation and they use var ShipsData which has different Values for baseFuelCapacity and baseCargoCapacity (the CargoCapacity already have the HyperSpace Bonus added so they display the correct cargo on the Page) and in the FuelCapacity computation they really use the baseFuelCapacity which is not equal the baseCargoCapacity in the HTML ShipsData variable. Check it on fleetDispatch page there is the JSON shipsData variable which contains all computed values for the Cargos. Thats the point where I know that the FuelCapacity is not increased by the Tech Hyperspace Technology.

```
  "204": {
    "id": 204,
    "name": "Light Fighter",
    "baseFuelConsumption": 20,
    "baseFuelCapacity": 50,
    "baseCargoCapacity": 62,
    "fuelConsumption": 20,
    "baseSpeed": 12500,
    "speed": 22500,
    "cargoCapacity": 62,
    "fuelCapacity": 50,
    "number": 1,
    "recycleMode": 0
  }
```

For example 1x LightFighter consumes 138 Fuel but its fuel Capacity is 50. Which leads into
138 / 50 * 100 = 276% (not equal) 138 / 62 * 100 = 222%

And this means it will not be increased by the HyperSpaceTechnology

![image](https://user-images.githubusercontent.com/21330562/131663044-ba5aed1c-0f2b-4c9e-979b-8d7cf0c3750b.png)

Reply:
I tried to send 1x LightFigher to a mission FuelConsumption 52 and CargoCapacity 62  so 52 is between 50 and 62. The mission fails with not enough cargo space available. So they use 50 for the FuelCapacity of the LightFighter internally. I remember as they added the Hyper Space Technology to increase Cargo Space the disconnected the Cargo from the FuelCapacity. And they only increase the Cargo and not the Fuel Capacity, which means the ships can have a higher range which was not wanted.

Reply:
Now I got you!
```
fuelCapacity := ogame.ShipsInfos{EspionageProbe: 1}.Cargo(ogame.Researches{}, true, false, false)
// fuelCapacity = 5
```